### PR TITLE
support imagePullSecret for pulling operator image.

### PR DESCRIPTION
**Pull request by morvencao**
Support set imagePullSecrets for operator init and dump commands by `--imagePullSecrets` flag.
Also fix the wrong operator dump golden file name.

[ ] Configuration Infrastructure
[ ] Docs
[ X ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: operator/cmd/mesh/operator-init.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ func addOperatorInitFlags(cmd *cobra.Command, args *operatorInitArgs) {
 	cmd.PersistentFlags().StringVar(&args.context, "context", "", ContextFlagHelpStr)
 	cmd.PersistentFlags().StringVar(&args.common.hub, "hub", hub, HubFlagHelpStr)
 	cmd.PersistentFlags().StringVar(&args.common.tag, "tag", tag, TagFlagHelpStr)
+	// nolint: lll
+	cmd.PersistentFlags().StringVar(&args.common.imagePullSecrets, "imagePullSecrets", "",
+		"The imagePullSecrets used to pull the operator image from private registry, could be secret list separated by comma, eg. 'imagePullSecret1,imagePullSecret2'")
```
Comment (shamsher31) [inline operator/cmd/mesh/operator-init.go:51]: Better to create  `imagePullSecretsHelpStr` and use here.

File: operator/cmd/mesh/root.go [hunk 2]
```diff
@@ -45,12 +45,14 @@ const (
...
-	ContextFlagHelpStr       = `The name of the kubeconfig context to use.`
-	HubFlagHelpStr           = `The hub for the operator controller image.`
-	TagFlagHelpStr           = `The tag for the operator controller image.`
+	installationCompleteStr = `Installation complete`
+	ForceFlagHelpStr        = `Proceed even with validation errors.`
+	KubeConfigFlagHelpStr   = `Path to kube config.`
+	ContextFlagHelpStr      = `The name of the kubeconfig context to use.`
+	HubFlagHelpStr          = `The hub for the operator controller image.`
+	TagFlagHelpStr          = `The tag for the operator controller image.`
+	ImagePullSecretsHelpStr = `The imagePullSecrets used to pull the operator image from private registry,
```
Comment (shamsher31) [inline operator/cmd/mesh/root.go:54]:
```suggestion
	ImagePullSecretsHelpStr = `The imagePullSecrets is used to pull the operator image from the private registry,
```

Comment (morvencao) [reply operator/cmd/mesh/root.go]: done, thanks

File: operator/cmd/mesh/operator-init.go (hunk 1 above)
Comment (morvencao) [reply operator/cmd/mesh/operator-init.go]: done, thanks

File: operator/cmd/mesh/operator_test.go [hunk 3]
```diff
@@ -76,7 +78,7 @@ func TestOperatorInit(t *testing.T) {
 			tag:               "1.2.3",
 			operatorNamespace: "operator-test-namespace",
 			watchedNamespaces: "istio-test-namespace1,istio-test-namespace2",
-			manifestsPath:     string(snapshotCharts),
+			manifestsPath:     string(liveCharts),
```
Comment (shamsher31) [inline operator/cmd/mesh/operator_test.go:81]: I think `liveCharts` read data from the `manifests` directory. So in that case this test will not read YAML from `testdata` directory?

File: operator/cmd/mesh/operator-init.go [hunk 4]
```diff
@@ -46,6 +46,7 @@ func addOperatorInitFlags(cmd *cobra.Command, args *operatorInitArgs) {
 	cmd.PersistentFlags().StringVar(&args.context, "context", "", ContextFlagHelpStr)
 	cmd.PersistentFlags().StringVar(&args.common.hub, "hub", hub, HubFlagHelpStr)
 	cmd.PersistentFlags().StringVar(&args.common.tag, "tag", tag, TagFlagHelpStr)
+	cmd.PersistentFlags().StringVar(&args.common.imagePullSecrets, "imagePullSecrets", "", ImagePullSecretsHelpStr)
```
Comment (howardjohn) [inline operator/cmd/mesh/operator-init.go:49]: isn't this a list?

Comment (howardjohn) [reply operator/cmd/mesh/operator-init.go:49]: Ah I see you split it. I think cobra allows StringSlice natively

Comment (morvencao) [reply operator/cmd/mesh/operator-init.go:49]: thanks for reminder, updated it to use `StringSlice`.

File: operator/cmd/mesh/operator_test.go (hunk 3 above)
Comment (morvencao) [reply operator/cmd/mesh/operator_test.go]: updated the snapshotCharts and use that.

**Review by shamsher31 — APPROVED** [review]
Thanks. LGTM.

**Comment by morvencao** [comment]
@istio/wg-environments-maintainers PTAL

**Review by ostromart — APPROVED** [review]

**Pull request merged**